maxfw/utils/image_functions.py

```diff
@@ -0,0 +1,524 @@
+
+    elif npimg.shape[2] == 2:
+        permitted_2_channel_modes = ['LA']
+        if mode is not None and mode not in permitted_2_channel_modes:
+            raise ValueError("Only modes {} are supported for 2D inputs".format(permitted_2_channel_modes))
+
+        if mode is None and npimg.dtype == np.uint8:
+            mode = 'LA'
+
+    elif npimg.shape[2] == 4:
+        permitted_4_channel_modes = ['RGBA', 'CMYK', 'RGBX']
+        if mode is not None and mode not in permitted_4_channel_modes:
+            raise ValueError("Only modes {} are supported for 4D inputs".format(permitted_4_channel_modes))
+
+        if mode is None and npimg.dtype == np.uint8:
+            mode = 'RGBA'
+    else:
+        permitted_3_channel_modes = ['RGB', 'YCbCr', 'HSV']
+        if mode is not None and mode not in permitted_3_channel_modes:
+            raise ValueError("Only modes {} are supported for 3D inputs".format(permitted_3_channel_modes))
+        if mode is None and npimg.dtype == np.uint8:
+            mode = 'RGB'
+
+    if mode is None:
+        raise TypeError('Input type {} is not supported'.format(npimg.dtype))
+
+    # Verify that the target mode exists
+    assert target_mode in [1, 'L', 'P', 'RGB', 'RGBA', 'CMYK', 'YCbCr', 'LAB', 'HSV', 'I', 'F', 'RGBX', 'RGBBa']
+
+    return Image.fromarray(npimg, mode=mode).convert(target_mode)
+
+
+def pil_to_array(pic):
+    assert _is_pil_image(pic), 'The input image for `PILtoarray` is not a PIL Image object.'
+    return np.array(pic)
+
+
+def normalize(img):
+    if type(img) is not np.ndarray:
+        img = np.array(img)
+    return img / (np.max(img) - np.min(img))
+
+
+def standardize(img, mean=None, std=None):
+    # ensure we are working with a numpy ndarray
+    if type(img) is not np.ndarray:
+        img = np.array(img)
+    img = img.astype(np.float64)
+
+    # check whether the image has channels
+    if img.ndim == 3:
+        # (this image has channels)
+        # calculate the number of channels
+        channels = img.shape[-1]
+
+        if mean is None:
+            # calculate channel-wise mean
+            mean = np.mean(img, axis=(0, 1), keepdims=True)
+        elif isinstance(mean, (int, float)):
+            # convert the number to an array
+            mean = np.array([mean] * channels).reshape((1, 1, channels))
+        elif isinstance(mean, Sequence):
+            # convert a sequence to the right dimensions
+            assert np.sum([not isinstance(x, (int, float)) for x in mean]) == 0, \
+                'The sequence `mean` can only contain numbers.'
+            assert len(mean) == channels, \
+                'The size of the `mean` array must correspond to the number of channels in the image.'
+            mean = np.array(mean).reshape((1, 1, channels))
+        else:
+            # if the mean is not a number or a sequence
+            raise ValueError('The value for `mean` should either be a number or an n-dimensional vector of numbers '
```

Comment:
@splovyt do you think we should raise TypeError instead of ValueError here?

Reply:
I will change it based on your suggestion

Reply:
@splovyt  I don't have any other comments. I can make this change btw, since it's a small change.

Reply:
Sure if it's easier for you